File: mambo-protocol/src/main/scala/org/mambo/protocol/types/game/context/fight/FightCommonInformations.scala





















// Generated on 05/03/2013 10:35:03
// Scala version by Blackrush
package org.mambo.protocol.types

import org.mambo.core.io._
import org.mambo.protocol.{Type => BaseType, _}
import org.mambo.protocol.enums._

trait FightCommonInformations extends TypeDeserializer {
  val typeId = 43

  protected trait Type extends BaseType with Serializable {
    val typeId = FightCommonInformations.typeId

    val fightId: Int
    val fightType: Byte
    val fightTeams: Seq[FightTeamInformations.TypeImpl]
    val fightTeamsPositions: Seq[Short]
    val fightTeamsOptions: Seq[FightOptionsInformations.TypeImpl]

    def serialize(buf: Buffer) {
      buf.writeInt(fightId)
      buf.writeByte(fightType)
      buf.writeUShort(fightTeams.length)
      for (entry <- fightTeams) {
        entry.serialize(buf)
      }
      buf.writeUShort(fightTeamsPositions.length)
      for (entry <- fightTeamsPositions) {
        buf.writeShort(entry)
      }
      buf.writeUShort(fightTeamsOptions.length)
      for (entry <- fightTeamsOptions) {
        entry.serialize(buf)
      }
    }

  }

  protected def create(fightId: Int, fightType: Byte, fightTeams: Seq[FightTeamInformations.TypeImpl], fightTeamsPositions: Seq[Short], fightTeamsOptions: Seq[FightOptionsInformations.TypeImpl]): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val fightId = buf.readInt()
    val fightType = buf.readByte()
    val fightTeams = {
      val builder = collection.immutable.Seq.newBuilder[FightTeamInformations.TypeImpl]
      for (i <- (0 to buf.readUShort())) {
        builder += FightTeamInformations.deserialize(buf)
      }
      builder.result()
    }
    val fightTeamsPositions = {
      val builder = collection.immutable.Seq.newBuilder[Short]
      for (i <- (0 to buf.readUShort())) {
        builder += buf.readShort()
      }
      builder.result()
    }
    val fightTeamsOptions = {
      val builder = collection.immutable.Seq.newBuilder[FightOptionsInformations.TypeImpl]
      for (i <- (0 to buf.readUShort())) {
        builder += FightOptionsInformations.deserialize(buf)
      }
      builder.result()
    }

    create(fightId, fightType, fightTeams, fightTeamsPositions, fightTeamsOptions)
  }

}

object FightCommonInformations extends FightCommonInformations {
  case class TypeImpl(fightId: Int, fightType: Byte, fightTeams: Seq[FightTeamInformations.TypeImpl], fightTeamsPositions: Seq[Short], fightTeamsOptions: Seq[FightOptionsInformations.TypeImpl]) extends Type
  override def create(fightId: Int, fightType: Byte, fightTeams: Seq[FightTeamInformations.TypeImpl], fightTeamsPositions: Seq[Short], fightTeamsOptions: Seq[FightOptionsInformations.TypeImpl]) = TypeImpl(fightId, fightType, fightTeams, fightTeamsPositions, fightTeamsOptions)

  def unapply(msg: TypeImpl) = TypeImpl.unapply(msg)
  def apply(fightId: Int, fightType: Byte, fightTeams: Seq[FightTeamInformations.TypeImpl], fightTeamsPositions: Seq[Short], fightTeamsOptions: Seq[FightOptionsInformations.TypeImpl]) = create(fightId, fightType, fightTeams, fightTeamsPositions, fightTeamsOptions)

  type Target = TypeImpl

}

File: mambo-protocol/src/main/java/org/mambo/protocol/types/FightCommonInformations.java


// Generated on 05/08/2013 19:38:00
package org.mambo.protocol.types;

import java.util.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class FightCommonInformations extends NetworkType {
    public static final short TYPE_ID = 43;
    
    @Override
    public short getTypeId() {
        return TYPE_ID;
    }
    
    public int fightId;
    public byte fightType;
    public FightTeamInformations[] fightTeams;
    public short[] fightTeamsPositions;
    public FightOptionsInformations[] fightTeamsOptions;
    
    public FightCommonInformations() { }
    
    public FightCommonInformations(int fightId, byte fightType, FightTeamInformations[] fightTeams, short[] fightTeamsPositions, FightOptionsInformations[] fightTeamsOptions) {
        this.fightId = fightId;
        this.fightType = fightType;
        this.fightTeams = fightTeams;
        this.fightTeamsPositions = fightTeamsPositions;
        this.fightTeamsOptions = fightTeamsOptions;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeInt(fightId);
        buf.writeByte(fightType);
        buf.writeUShort(fightTeams.length);
        for (FightTeamInformations entry : fightTeams) {
            entry.serialize(buf);
        }
        buf.writeUShort(fightTeamsPositions.length);
        for (short entry : fightTeamsPositions) {
            buf.writeShort(entry);
        }
        buf.writeUShort(fightTeamsOptions.length);
        for (FightOptionsInformations entry : fightTeamsOptions) {
            entry.serialize(buf);
        }
    }
    
    @Override
    public void deserialize(Buffer buf) {
        fightId = buf.readInt();
        fightType = buf.readByte();
        if (fightType < 0)
            throw new RuntimeException("Forbidden value on fightType = " + fightType + ", it doesn't respect the following condition : fightType < 0");
        int limit = buf.readUShort();
        fightTeams = new FightTeamInformations[limit];
        for (int i = 0; i < limit; i++) {
            fightTeams[i] = new FightTeamInformations();
            fightTeams[i].deserialize(buf);
        }
        limit = buf.readUShort();
        fightTeamsPositions = new short[limit];
        for (int i = 0; i < limit; i++) {
            fightTeamsPositions[i] = buf.readShort();
        }
        limit = buf.readUShort();
        fightTeamsOptions = new FightOptionsInformations[limit];
        for (int i = 0; i < limit; i++) {
            fightTeamsOptions[i] = new FightOptionsInformations();
            fightTeamsOptions[i].deserialize(buf);
        }
    }
    
}
